packages/jest-transform/src/ScriptTransformer.ts

```diff
@@ -659,16 +659,18 @@ const calcIgnorePatternRegExp = (config: Config.ProjectConfig) => {
 };
 
 const calcTransformRegExp = (config: Config.ProjectConfig) => {
-  if (!config.transform.length) {
+  const transform = config.transform || [];
```

Comment:
the `|| []` shouldn't be needed

Reply:
My comment about this disappeared: it's b/c in a number of tests we do `const { ... } as Config.ProjectConfig`, meaning `config.transform` is not always defined.

It's either this, or update every test config. This was the quickest way to get things passing, while waiting to see what you guys wanted to do 🙂

Reply:
Updating test config is more correct - what they're doing is a type error and we don't need the (albeit negligible) overhead in production code.

We have helpers for creating project config in tests here: https://github.com/facebook/jest/blob/012472b6ca4488e55235d0e31e0c760364cc07a2/TestUtils.ts#L142-L156. Can probably use that?

Reply:
> Updating test config is more correct

That was my thoughts too - as you probably noticed in my work for `eslint-plugin-jest`, I like using `any` & `as` as little as possible, since they make it very hard to do global refactors (among other reasons) 😬

I'll hunt down all the erroneous configs that need to change for this PR.

Let me know if you'd like me to do a sweep over all the other tests for casting configs, & replace them w/ `makeProjectConfig` via separate PR :)

Reply:
> Let me know if you'd like me to do a sweep over all the other tests for casting configs, & replace them w/ `makeProjectConfig` via separate PR :)

That would be absolutely lovely!

Reply:
> That would be absolutely lovely!

I checked - these are the only tests that do this :(

Still, I'll hunt around later for anywhere I can strengthen the typings.

I'll start by running `tsc --noEmit` over everything, b/c there's definitely some tests that error under TypeScript i.e `SearchSource.test` (missing `maxWorkers` property from `InitialOptions`).

For now this should be resolved for this PR.